### ISSUE TITLE
Add automatic one-hot detection in inverse transform

### ISSUE THREAD

**Issue by maxsch3** [maxsch3/keras-batchflow#89]
Keras/TF always return categories as one-hot encoding although input labels can be encoded as integers and encoder expects integer in inverse_transform

It is convenient to detect and fix those scenarios by converting one-hot encoded data to integers using argmax.

**Comment by maxsch3** [comment]
Because sklearn encoders (default scenario) do not indicate expected shape of the input to inverse_transform, this cannot be done well. 
For example, LabelEncoder and LabelBinarizer encode categorical features in integers and one-hot encoded matrix. If they used for encoding output labels in a tensforflow model, it will output one-hot encoded output in any scenario (although in LabelEncoder scenario one would use sparse_categorical_crossentropy). When sending predictions generated by the model, it is no good way to determine what format the encoder expects in its inverse_transform. There are two not-so-good ways to determine that:

1.  Using Try-catch, but unfortunately sklearn throws generic ValueError exception if the input is in wrong format. This way it is not possible to catch specific error and there is a risk to catch confuse causes of the exception
2. Using name of the encoder, but thats effectively a hardcode, which I dont want to do

It is not doable as things stand at the moment
